### PR TITLE
258 cicd

### PR DESCRIPTION
**Pull request by akirakakar**
# Type of PR

- [ ] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## Purpose

- Add cicd for infra 

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Author pre-publish checklist:

- [ ] Added test to prove my fix is effective or new feature works
- [x] No PII in logs
- [ ] Made corresponding changes to the documentation

### CODE REVIEW

File: e2e_samples/dataset_versioning/infra/setup.sh [hunk 1]
```diff
@@ -1,96 +1,81 @@
...
-cat << EOF
-
-# This is required for the next step in the ReadMe (Step #3)
-# Initialize the Terraform state and store it in the Azure resources
-az login
-cd terraform/live/dev
-terraform init -backend-config="storage_account_name=${TF_STATE_STORAGE_ACCOUNT_NAME}dev" -backend-config="container_name=$TF_STATE_CONTAINER_NAME" -backend-config="access_key=\$(az keyvault secret show --name tfstate-storage-key-dev --vault-name $KEYVAULT_NAME --query value -o tsv)" -backend-config="key=terraform.tfstate"
-
-EOF
+while true; do
```
Comment (shervyna) [inline e2e_samples/dataset_versioning/infra/setup.sh:3]: if you change this one, remember to update the commands in readme.md as well :)
also I forgot to mention in your previous PR, please update the terraform state list in readme.md for the newly created resources. (or we can just delete that section if it is not needed :D

File: e2e_samples/dataset_versioning/infra/terraform/.azdo/iac-template.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+- name: buildingEnvironment
+  type: string
+  default: staging
+jobs:
+- deployment: IaC
+  displayName: IaC
+  pool:
+    vmImage: 'Ubuntu-latest'
+  variables:
+  - group: terraform-secret-${{ parameters.buildingEnvironment }}
```
Comment (shervyna) [inline e2e_samples/dataset_versioning/infra/terraform/.azdo/iac-template.yml:17]: you might need to add readme about how to create variable groups and enviroments in ADO? I think in the temperature-events folder there might be some reusable parts.

Comment (akirakakar) [reply e2e_samples/dataset_versioning/infra/terraform/.azdo/iac-template.yml:17]: Add some details on pipeline

File: e2e_samples/dataset_versioning/infra/setup.sh (hunk 1 above)
Comment (akirakakar) [reply e2e_samples/dataset_versioning/infra/setup.sh:3]: done.

File: e2e_samples/dataset_versioning/infra/README.md [hunk 3]
```diff
@@ -69,6 +71,11 @@ terraform plan -var "name=myapp" -out=/tmp/myapp
 terraform apply /tmp/myapp
 ```
 
+### Other options
```
Comment (shervyna) [inline e2e_samples/dataset_versioning/infra/README.md:74]: nit: maybe add CICD on the section title to be more clear?
also, we have this line:
  environment: ${{ parameters.buildingEnvironment }}
If people dont set the env, will it throw error? if so, let's mention setup enviroment here as well :)

Comment (akirakakar) [reply e2e_samples/dataset_versioning/infra/README.md:74]: Let me update the whole readme as different issue for cleaning up.

**Pull request merged**